sys/include/net/ng_netdev.h

```diff
@@ -0,0 +1,169 @@
+     * @return              -ENOMSG if pkt is invalid
+     */
+    int (*send_data)(ng_netdev_t *dev, ng_pktsnip_t *pkt);
+
+    /**
+     * @brief   Registers an event callback to a given network device
+     *
+     * @param[in] dev       network device descriptor
+     * @param[in] cb        function that is called on events
+     *
+     * @return              0 on success
+     * @return              -ENODEV if *dev* is invalid
+     * @return              -ENOBUFS if maximum number of callbacks is exceeded
+     */
+    int (*add_event_callback)(ng_netdev_t *dev, ng_netdev_event_cb_t cb);
+
+    /**
+     * @brief   Unregisters an event callback from a given network device
+     *
+     * @param[in] dev       network device descriptor
+     * @param[in] cb        callback function
+     *
+     * @return              0 on success
+     * @return              -ENODEV if *dev* is invalid
+     * @return              -ENOPKG if callback was not registered
+     */
+    int (*rem_event_callback)(ng_netdev_t *dev, ng_netdev_event_cb_t cb);
+
+    /**
+     * @brief   Get an option value from a given network device
+     *
+     * @param[in] dev           network device descriptor
+     * @param[in] opt           option type
+     * @param[out] value        pointer to store the option's value in
+     * @param[in,out] value_len the length of *value*. Must be initialized to
+     *                          the available space in *value* on call.
+     *
+     * @return              0 on success
+     * @return              -ENODEV if *dev* is invalid
+     * @return              -ENOTSUP if *opt* is not supported
+     * @return              -EOVERFLOW if available space in *value* given in
+     *                      *value_len* is too small to store the option value
+     */
+    int (*get)(ng_netdev_t *dev, ng_netconf_opt_t opt,
+                      void *value, size_t *value_len);
+
+    /**
+     * @brief   Set an option value for a given network device
+     *
+     * @param[in] dev       network device descriptor
+     * @param[in] opt       option type
+     * @param[in] value     value to set
+     * @param[in] value_len the length of *value*
+     *
+     * @return              0 on success
+     * @return              -ENODEV if *dev* is invalid
+     * @return              -ENOTSUP if *opt* is not supported
+     * @return              -EINVAL if *value* is invalid
+     */
+    int (*set)(ng_netdev_t *dev, ng_netconf_opt_t opt,
+               void *value, size_t value_len);
+
+    /**
+     * @brief   This function is called by a MAC layer when a message of type
+     *          NETDEV_MSG_EVENT_TYPE was received
+     *
+     * @param[in] dev           network device descriptor
+     * @param[in] event_type    event type, given by @ref msg_t::value in the
+     *                          received message
+     */
+    void (*isr_event)(ng_netdev_t *dev, uint16_t event_type);
+} ng_netdev_driver_t;
+
+struct ng_netdev_t {
```

Comment:
if you look some lines up in the code, you see it is typedefed... The forward declaration is needed as the type is used in `ng_netdev_driver_t`...